src/status_im/contexts/wallet/collectible/events.cljs

```diff
@@ -166,39 +160,58 @@
    (let [request-id               0
          collectible-id-converted (cske/transform-keys transforms/->PascalCaseKeyword collectible-id)
          data-type                (collectible-data-types :details)
          request-params           [request-id [collectible-id-converted] data-type]]
-     {:db (assoc-in db [:wallet :last-collectible-aspect-ratio] aspect-ratio)
+     {:db (assoc-in db
+           [:wallet :ui :collectible]
+           {:details        collectible
+            :aspect-ratio   aspect-ratio
+            :gradient-color gradient-color})
       :fx [[:json-rpc/call
             [{:method   "wallet_getCollectiblesByUniqueIDAsync"
               :params   request-params
               :on-error (fn [error]
                           (log/error "failed to request collectible"
-                                     {:event  :wallet/get-collectible-details
+                                     {:event  :wallet/navigate-to-collectible-details
                                       :error  error
-                                      :params request-params}))}]]]})))
+                                      :params request-params}))}]]
+           ;; We delay the navigation because we need re-frame to update the DB on time.
+           ;; By doing it, we skip a blink while visiting the collectible detail page.
+           [:dispatch-later
+            {:ms       1
+             :dispatch [:navigate-to :screen/wallet.collectible]}]]})))
+
+(defn- keep-not-empty-value
+  [old-value new-value]
+  (if (or (string/blank? new-value) (nil? new-value))
+    old-value
+    new-value))
+
+(def merge-skipping-empty-values (partial merge-with keep-not-empty-value))
 
 (rf/reg-event-fx
  :wallet/get-collectible-details-done
- (fn [_ [{:keys [message]}]]
-   (let [response               (cske/transform-keys transforms/->kebab-case-keyword
-                                                     (transforms/json->clj message))
-         {:keys [collectibles]} response
-         collectible            (first collectibles)]
+ (fn [{db :db} [{:keys [message]}]]
+   (let [response                      (cske/transform-keys transforms/->kebab-case-keyword
+                                                            (transforms/json->clj message))
+         {[collectible] :collectibles} response]
      (if collectible
-       {:fx [[:dispatch [:wallet/store-last-collectible-details collectible]]]}
+       {:db (update-in db [:wallet :ui :collectible :details] merge-skipping-empty-values collectible)}
        (log/error "failed to get collectible details"
                   {:event    :wallet/get-collectible-details-done
                    :response response})))))
 
 (rf/reg-event-fx
- :wallet/clear-last-collectible-details
+ :wallet/clear-collectible-details
  (fn [{:keys [db]}]
-   {:db (update-in db [:wallet] dissoc :last-collectible-details :last-collectible-aspect-ratio)}))
+   {:db (update-in db [:wallet :ui] dissoc :collectible)}))
```

Comment:
Events were renamed to remove the word `last`, that `last` word is misleading, since we don't have multiple collectibles or a first one.

Now the events and keys are just referring to it as `:collectible`. and the route in the dB is `:wallet :ui :collectible`